docs/typescript/reference/client.md

```diff
@@ -0,0 +1,152 @@
+
+Enqueues a workflow, similar to [passing a queue name to `startWorkflow`](./transactapi/dbos-class.md#starting-background-workflows)
+or using [`DBOS.withWorkflowQueue`](./transactapi/dbos-class.md#using-workflow-queues). 
+Like `startWorkflow`, the `enqueue` method returns a `WorkflowHandle` that you can use to retrieve the workflow results 
+asynchronously from an external application.
+
+When enqueuing a workflow from within a DBOS application, the workflow and queue metadata can be retrieved automatically.
+However, since `DBOSClient` runs outside the DBOS application, the metadata must be specified explicitly.
+This metadata includes:
+
+* `workflowName`: The name of the workflow method being enqueued.
+* `workflowClassName`: The name of the class the workflow method is a member of.
+* `queueName`: The name of the [WorkflowQueue](./transactapi/workflow-queues#class-workflowqueue) to enqueue the workflow on.
+* `appVersion`: The version of your application that should process this workflow. If left undefined, it will be updated to the current version when the workflow is first dequeued. Please see [Managing Application Versions](../../production/self-hosting/workflow-recovery#managing-application-versions) for more information.
+* `workflowID`: The unique ID for the enqueued workflow. DBOS Client will generate a [UUID](https://en.wikipedia.org/wiki/Universally_unique_identifier) if left undefined. Please see [Workflow IDs and Idempotency](../tutorials/workflow-tutorial#workflow-ids-and-idempotency) for more information.
+
+In addition to the `EnqueueOptions` described above, you must also provide the workflow arguments to `enqueue`. 
+These are passed to `enqueue` after the initial `EnqueueOptions` parameter.
+By default, the workflow arguments are not type checked. 
+However, you can achieve type safety for the arguments and the returned `WorkflowHandle` by specifying the function type parameter when calling enqueue.
+
+:::tip
+Note, DBOS Client is designed to work independently of your DBOS application code.
+You do not need your DBOS workflow function's implementation to make `enqueue` type safe.
+You only need the type declaration for the function.
+
+Type declarations are automatically generated by the TypeScript compiler during compilation of your DBOS application.
+You can copy or import the declaration from your application's [generated declaration file (aka.d.ts file)](https://www.typescriptlang.org/docs/handbook/declaration-files/introduction.html).
+::: 
+
+Example:
+
+```ts
+// TypeScript type declaration for our sample app workflow
+declare class DocumentDetective {
+    static indexDocument(url: string): Promise<number>
+}
+
+// create a helper type for the workflow we are enqueuing 
+type IndexDocument = typeof DocumentDetective.indexDocument;
+
+// type safe arguments for enqueue call by use of 
+// the helper type + Parameters utility type
+const documentWFID = //application specific code to generate an unique ID
```

Comment:
We can shorten this example a bit to emphasize how simple clients are, maybe don't use WFID and consolidate the last few comments.